src/components/user/LoginRedirections.js

```diff
@@ -0,0 +1,43 @@
+import {Link, useLocation} from "react-router-dom";
+import {routes} from "../../config/routes";
+import {Button, Divider, Icon} from "semantic-ui-react";
+import React from "react";
+import {useTranslation} from "react-i18next";
+import GoogleLoginOption from "./GoogleLogin";
+
+
+const OtherLoginOptions = ({setError, setLoading}) => {
+  const { t } = useTranslation();
+  let location = useLocation();
+
+  const links = [
+    {link: routes.login(), icon: "sign-in", text: t("login.actions.login"), },
+    {link: routes.register(), icon: "signup", text: t("login.actions.register")},
+    {link: routes.login_guest(), icon: "user secret", text: t("login.actions.guest_login")}
+  ];
+
+  return (
+    <>
+      <Divider/>
+      {links.map((link) => {
+        if (link.link !== location.pathname)
+          return <OtherLoginOption link={link} key={link.link}/>
+      })}
+      <GoogleLoginOption setError={setError} setLoading={setLoading}/>
+    </>
+  );
+};
+
+
+const OtherLoginOption = ({link }) => {
+  return (
+    <Link to={link.link}>
```

Comment:
remember to pass locations state here so we can later redirect user to the initial page:
```
import { Link, useLocation } from "react-router-dom";
const location = useLocation();

....


           <Link
              to={{
                pathname: routes.login(),
                state: location.state,
              }}
            >
              <Button content={t("login.actions.login")} />
            </Link>
```

Reply:
👍